`make clippy` failed

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?

git commit SHA: [a54bb5a7f97829afedbc042ef1284cddf1deb5bf](https://github.com/tikv/tikv/commit/a54bb5a7f97829afedbc042ef1284cddf1deb5bf)

### What operating system and CPU are you using?

```console
> uname -a
Darwin zuowuhuideMacBook-Pro.local 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64
```

### Steps to reproduce
```console
> make clippy
......
 advisories FAILED: 1 errors, 0 warnings, 18 notes
           bans ok: 0 errors, 71 warnings, 14 notes
       licenses ok: 0 errors, 0 warnings, 623 notes
        sources ok: 0 errors, 0 warnings, 27 notes
make: *** [clippy] Error 1
```

### What did you expect?
`make clippy` run succeed.

### What did happened?

It failed.

--- a/clippy.toml
+++ b/clippy.toml
@@ -51,6 +51,14 @@ reason = """
 X509StoreRef::objects is unsound, see RUSTSEC-2020-0071
 """
 
+# See more about RUSTSEC-2025-0004 in deny.toml.
+[[disallowed-methods]]
+path = "openssl::ssl::select_next_proto"
+reason = """
+openssl::ssl::select_next_proto may return a buffer use after free, \
+see RUSTSEC-2025-0004
+"""
+
 # See more about RUSTSEC-2024-0357 in deny.toml.
 [[disallowed-types]]
 path = "openssl::bio::MemBio"

--- a/deny.toml
+++ b/deny.toml
@@ -73,6 +73,13 @@ ignore = [
     #
     # TODO: Upgrade clap to v4.x.
     "RUSTSEC-2021-0145",
+    # Ignore RUSTSEC-2025-0004, as it will trigger a recursive upgrade of OpenSSL
+    # to version 3.x.
+    #
+    # NB: Upgrading openssl the version >= 0.10.70 do fix the issue but it
+    # also upgrade the OpenSSL to v3.x which causes performance degradation.
+    # See https://github.com/openssl/openssl/issues/17064
+    "RUSTSEC-2025-0004",
 ]
 
 # TiKV is licensed under Apache 2.0, according to ASF 3RD PARTY LICENSE POLICY,

